assembl/views/api2/auth.py

```diff
@@ -452,21 +475,36 @@ def assembl_register_user(request):
         # Find agent account to avoid duplicates!
         if session.query(AbstractAgentAccount).filter_by(
                 email_ci=email).count():
-            errors.add_error(localizer.translate(_(
-                "We already have a user with this email.")),
-                ErrorTypes.EXISTING_EMAIL,
-                HTTPConflict.code)
+            if not discussion.preferences['generic_auth_errors']:
+                errors.add_error(localizer.translate(_(
+                    "We already have a user with this email.")),
+                    ErrorTypes.EXISTING_EMAIL,
+                    HTTPConflict.code)
+            else:
+                errors.add_error(localizer.translate(_(
+                    generic_error_message)),
+                    ErrorTypes.EXISTING_EMAIL,
+                    HTTPConflict.code)
+                logger.error("We already have a user with this email %s" % email)
+
     if not email:
         errors.add_error(localizer.translate(_("No email.")),
                          ErrorTypes.INVALID_EMAIL)
     username = json.get('username', None)
     if username:
         if session.query(Username).filter(
                 func.lower(Username.username) == username.lower()).count():
-            errors.add_error(localizer.translate(_(
-                "We already have a user with this username.")),
-                ErrorTypes.EXISTING_USERNAME,
-                HTTPConflict.code)
+            if not discussion.preferences['generic_auth_errors']:
```

Comment:
You've made the comparison `not discussion.preference['generic_auth_errors']` multiple times. Could create a variable use `generic_error = discussion.preference['generic_auth_errors']` and do a comparison this way, if you wish.